core/core-persistence/core-persistence-infinispan/src/main/java/ai/wanaku/core/persistence/infinispan/InfinispanToolReferenceRepository.java

```diff
@@ -26,4 +30,13 @@ protected Class<ToolReference> entityType() {
     @Override
     protected String newId() {
         return UUID.randomUUID().toString();
-    }}
+    }
+
+    @Override
+    public List<ToolReference> findByName(String name) {
+      Query<ToolReference> query = cacheManager.getCache(entityName()).query("from ai.wanaku.api.types.ToolReference t where t.name = :name");
+      query.setParameter("name", name);
+      return query.execute().list();
+    }
```

Comment:
I like this. I wasn't aware we could do it. There's a few places where we might benefit from this approach.